### PR TITLE
[format] black formatting

### PR DESCRIPTION
**Pull request by daniel-goldstein**
@danking Here's what it looks like running black on `ci.py` and `address.py` with only customizations being a line length of 120 (I personally prefer shorter but that's what we have currently) and no single quote -> double quote changes (`--skip-string-normalization`).

It is _super_ opinionated, so you have to like the defaults or be willing to roll with it on some calls. It took a lot of [discourse](https://github.com/psf/black/issues/118) just to allow single-quoting strings, but a commit hook to auto-format service code seems useful.

### CODE REVIEW

**Comment by danking** [comment]
I have no qualms with this.

I do hate double quotes, but I guess he's right that black could just reformat my single quotes into double quotes.

**Comment by danking** [comment]
Looks like I can just enable this with elpy in Emacs. What else would we do? Set a git hook?

**Comment by danking** [comment]
We can probably turn flake8 off. I don't know what good it provides over pylint and mypy.

**Comment by daniel-goldstein** [comment]
Ya I'll see if `pylint` and `mypy` give good enough coverage, and whether they fight with `black` over anything. Something useful from that thread was they point out how a big formatting commit ruins git blame unless you account for it.

I think just a pre-commit hook is good, anyone who wants to set it up in their editor can do so fairly easily.

**Comment by vladsaveliev** [comment]
Hi,

I'm just curious if you tried black on a code that uses Hail query. As far as I see, PEP8 conflicts with the code style adopted in the Hail docs, e.g. black would remove spaces in named function arguments:

```
- mt = mt.annotate_entries(GT = lgt_to_gt(mt.LGT, mt.LA))
+ mt = mt.annotate_entries(GT=lgt_to_gt(mt.LGT, mt.LA))
```

On the other hand, query can be seen as a DSL on top of Python, so the same guidelines probably don't need to be applied to it. Wondering if you had thoughts about lining the query code? We will be writing a lot of that in the nearest future in the Centre for Population Genomics, and would love to set up some style checks, or even automate that with a tool like black or prettier.

And on black - are you considering automating code refactoring with black as part of the CI? Or you wanted to just do checks, alongside with pylint?

Vlad

**Comment by tpoterba** [comment]
> On the other hand, query can be seen as a DSL on top of Python, so the same guidelines probably don't need to be applied to it

You've hit exactly on the center of a longstanding discussion on the team with no definitive resolution 😄

**Comment by daniel-goldstein** [comment]
Great question! I have not thought to run black on query and agree with your point about it maybe not being a good match. The intention here was just for formatting internal python services. Even so I don't love quite how restrictive black is, but it does seem like the most popular formatter out there now.

**Comment by danking** [comment]
FWIW, when I write hail code, I usually write it in PEP8 style because my editor complains. I've grown accustomed to it. I am in favor of blackening everything for consistency.

Re: CI, I think we'll ban (non-hail-query) code that doesn't conform to Black and developers are then incentivized to add a git-hook or editor modification that runs black.

**Comment by daniel-goldstein** [comment]
I'm going to change this branch to instead add `pre-commit` with `black` as one of its commit hooks. Developers can choose to `pre-commit install` which will set up useful hooks and run `black` over any modified files for directories that opt into it (which I'll start as just CI).

File: hail/python/dev-requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 flake8==3.7.8
 mypy==0.780
 pylint==2.6.0
+pre-commit==2.9.2
```
Comment (danking) [inline hail/python/dev-requirements.txt:4]: should `black` be included here?

**Review by danking — CHANGES_REQUESTED** [review]

**Review by danking — APPROVED** [review]
Disregard, I understand now.

**Review by danking — CHANGES_REQUESTED** [review]
Can we also enforce this with a build.yaml step that verifies that `ci/` is blackened?

**Comment by daniel-goldstein** [comment]
There's a `--check` flag, I added it to the existing ci `check`. Had to add `black` to the dev dependencies for that.

**Review by danking — APPROVED** [review]

**Comment by danking** [comment]
Remove draft whenever you're ready. Looks good to me.

**Pull request merged**